src/popup/popup.js

```diff
@@ -111,6 +118,33 @@ elements.clearFiltersButton.addEventListener('click', () => {
   elements.searchButton.click();
 });
 
+// a function to fetch licencse explanation
+const tooltiGen = () => {
+  const pathId = this.id;
+  const heading = tooltipInfo[pathId].tooltipHeading;
+  const content = tooltipInfo[pathId].tooltipContent;
```

Comment:
If I replace all individual selectors with a single selector using a common id and `querySelectorAll` how will I be able to access the respective heading and content for each tooltip? (since indiviudal IDs were corresponding to the object containing the information)

Reply:
The common class (not id) is an addition and not a replacement for selecting all the required DOM elements at once. When you will be looping over them, you can get the IDs through the already present id. Also, instead of using `id`, I would recommend you to use [HTML5 data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) for storing the license name as it would be more semantical.